File: python/socketd/socketd/transport/server/ServerConfig.py
from socketd.transport.core.impl.ConfigBase import ConfigBase


class ServerConfig(ConfigBase):
    def __init__(self, schema):
        super().__init__(False)

        self.__schema = schema

        if schema.startswith("sd:"):
            schema = schema[3:]

        self.__schemaCleaned = schema;
        self.__host = ""
        self.__port = 8602

    def get_schema(self):
        return self.__schema

    def get_host(self) -> str:
        return self.__host

    def host(self, host):
        self.__host = host
        return self;

    def get_port(self) -> int:
        return self.__port

    def port(self, port):
        self.__port = port
        return self

    def get_local_url(self) ->str:
        if self.__host:
            return f"sd:{self.__schemaCleaned}://{self.__host}:{self.__port}"
        else:
            return f"sd:{self.__schemaCleaned}://127.0.0.1:{self.__port}"

    def __str__(self):
        return f"ServerConfig{{schema='{self.__schemaCleaned}', " \
               f"charset='{self._charset}', "\
               f"host='{self.__host}', " \
               f"port={self.__port}, " \
               f"ioThreads='{self._ioThreads}', " \
               f"codecThreads={self._codecThreads}, " \
               f"exchangeThreads={self._exchangeThreads}, " \
               f"idleTimeout={self._idleTimeout}, "\
               f"requestTimeout={self._requestTimeout}, " \
               f"streamTimeout={self._streamTimeout}, " \
               f"readBufferSize={self.__read_buffer_size}, " \
               f"writeBufferSize={self.__write_buffer_size}, " \
               f"maxUdpSize={self._maxUdpSize}}}"

File: javascript/socketd/src/socketd/transport/server/ServerConfig.ts
import {ConfigBase} from "../core/Config";

export class ServerConfig extends ConfigBase {
    private readonly _schema: string;
    private readonly _schemaCleaned: string;

    //主机名
    private _host: string;
    //端口
    private _port: number;
    //http server
    private _httpServer: any;

    constructor(schema: string) {
        super(false);

        this._schema = schema;

        //支持 sd: 开头的架构
        if (schema.startsWith("sd:")) {
            schema = schema.substring(3);
        }

        this._schemaCleaned = schema;

        this._host = "";
        this._port = 8602;
    }


    /**
     * 获取协议架构
     */
    getSchema(): string {
        return this._schema;
    }

    /**
     * 获取主机
     */
    getHost(): string {
        return this._host;
    }

    /**
     * 获取 http-server
     * */
    getHttpServer(): any {
        return this._httpServer;
    }

    /**
     * 配置 http-server（可共用端口）
     * */
    httpServer(httpServer: any): this {
        this._httpServer = httpServer;
        return this;
    }

    /**
     * 配置主机
     */
    host(host: string): this {
        this._host = host;
        return this;
    }

    /**
     * 获取端口
     */
    getPort(): number {
        return this._port;
    }

    /**
     * 配置端口
     */
    port(port: number): this {
        this._port = port;
        return this;
    }

    /**
     * 获取本机地址
     */
    getLocalUrl(): string {
        if (this._host) {
            return "sd:" + this._schemaCleaned + "://" + this._host + ":" + this._port;
        } else {
            return "sd:" + this._schemaCleaned + "://127.0.0.1:" + this._port;
        }
    }

    toString(): string {
        return "ServerConfig{" +
            "schema='" + this._schemaCleaned + '\'' +
            ", charset=" + this._charset +
            ", host='" + this._host + '\'' +
            ", port=" + this._port +
            ", ioThreads=" + this._ioThreads +
            ", codecThreads=" + this._codecThreads +
            ", exchangeThreads=" + this._exchangeThreads +
            ", idleTimeout=" + this._idleTimeout +
            ", requestTimeout=" + this._requestTimeout +
            ", streamTimeout=" + this._streamTimeout +
            ", readBufferSize=" + this._readBufferSize +
            ", writeBufferSize=" + this._writeBufferSize +
            ", maxUdpSize=" + this._maxUdpSize +
            '}';
    }
}